js/static/audioProcessor.js

```diff
@@ -0,0 +1,38 @@
+class AudioDataExtractor extends AudioWorkletProcessor {
```

Comment:
read more [here](https://developer.mozilla.org/en-US/docs/Web/API/AudioWorkletProcessor)

Reply:
why a js file here and not a typescript one?